### PR TITLE
BUG: Support timespec argument in Timestamp.isoformat() (#26131)

### LINKED ISSUE #26131: pd.Timestamp.isoformat does not implement keyword timespec
## Code Sample

**[IN]**
```python
import pandas as pd
now = pd.Timestamp.utcnow()
help(now.isoformat)
```
**[OUT]**
    Help on method isoformat in module pandas._libs.tslibs.timestamps:
    
    isoformat(sep='T') method of pandas._libs.tslibs.timestamps.Timestamp instance
        [sep] -> string in ISO 8601 format, YYYY-MM-DDT[HH[:MM[:SS[.mmm[uuu]]]]][+HH:MM].
        sep is used to separate the year from the time, and defaults to 'T'.
        timespec specifies what components of the time to include (allowed values are 'auto', 'hours', 'minutes', 'seconds', 'milliseconds', and 'microseconds').
    
    

**[IN]**
```python
now.isoformat(timespec="seconds")
```

**[OUT]**
    ---------------------------------------------------------------------------

    TypeError                                 Traceback (most recent call last)

    <ipython-input-8-3626478eeb62> in <module>
    ----> 1 now.isoformat(timespec="seconds")
    

    pandas\_libs\tslibs\timestamps.pyx in pandas._libs.tslibs.timestamps.Timestamp.isoformat()
    

    TypeError: isoformat() got an unexpected keyword argument 'timespec'

**[IN]**
```python
now.to_pydatetime().isoformat(timespec="seconds")
```

**[OUT]**
    '2019-04-18T07:38:50+00:00'

## Problem description

The help method has given me the feeling that I can use timespec just like I can with standard datetime objects (since python 3.6).

## Expected Output

We could mimic standard datetime.isoformat behavior, or at least remove this false information from the help message.

## Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: fr_FR.UTF-8
LOCALE: None.None

pandas: 0.24.2
pytest: 4.4.1
pip: 19.0.3
setuptools: 41.0.0
Cython: None
numpy: 1.16.2
scipy: 1.2.1
pyarrow: None
xarray: None
IPython: 7.4.0
sphinx: 2.0.1
patsy: 0.5.1
dateutil: 2.8.0
pytz: 2019.1
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.3
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: 4.3.3
bs4: 4.7.1
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10.1
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by marius-mather**
- [x] closes #26131
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

I have just tried to support the same behaviour as `datetime.datetime()`, without adding the additional `"nanoseconds"` option, but this could be handled as well.

### CODE REVIEW

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -607,6 +607,7 @@ Datetimelike
 - Bug in :meth:`Series.isin` with ``datetime64[ns]`` dtype and :meth:`.DatetimeIndex.isin` failing to consider timezone-aware and timezone-naive datetimes as always different (:issue:`35728`)
 - Bug in :meth:`Series.isin` with ``PeriodDtype`` dtype and :meth:`PeriodIndex.isin` failing to consider arguments with different ``PeriodDtype`` as always different (:issue:`37528`)
 - Bug in :class:`Period` constructor now correctly handles nanoseconds in the ``value`` argument (:issue:`34621` and :issue:`17053`)
+- Bug in :meth:`Timestamp.isoformat`, now handles the ``timespec`` argument from the base :class:``datetime`` class (:issue:`26131`)
```
Comment (arw2019) [inline doc/source/whatsnew/v1.2.0.rst:610]: The issue is tagged as enhancement so this probably should be moved there 

Also I would expect this to target v1.3 not v1.2

File: pandas/tests/scalar/timestamp/test_formats.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from pandas import Timestamp
+
+
+def test_isoformat():
+    ts = Timestamp(
```
Comment (arw2019) [inline pandas/tests/scalar/timestamp/test_formats.py:5]: Does this handle a nanosecond argument?

**Review by arw2019 — COMMENTED** [review]
Thanks for the PR!

Looks good, some comments

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (jreback) [reply doc/source/whatsnew/v1.2.0.rst:610]: yeah pls move this to 1.3 under other enhancements

File: pandas/_libs/tslibs/timestamps.pyx [hunk 3]
```diff
@@ -609,8 +609,8 @@ cdef class _Timestamp(ABCTimestamp):
     # -----------------------------------------------------------------
     # Rendering Methods
 
-    def isoformat(self, sep: str = "T") -> str:
-        base = super(_Timestamp, self).isoformat(sep=sep)
+    def isoformat(self, sep: str = "T", timespec: str = "auto") -> str:
+        base = super(_Timestamp, self).isoformat(sep=sep, timespec=timespec)
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx:613]: can you also add a doc-string here

File: pandas/_libs/tslibs/timestamps.pyx [hunk 4]
```diff
@@ -609,8 +609,8 @@ cdef class _Timestamp(ABCTimestamp):
     # -----------------------------------------------------------------
     # Rendering Methods
 
-    def isoformat(self, sep: str = "T") -> str:
-        base = super(_Timestamp, self).isoformat(sep=sep)
+    def isoformat(self, sep: str = "T", timespec: str = "auto") -> str:
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx:612]: is 'auto' the typical name for this?

File: pandas/tests/scalar/timestamp/test_formats.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from pandas import Timestamp
+
+
+def test_isoformat():
```
Comment (jreback) [inline pandas/tests/scalar/timestamp/test_formats.py:4]: can you locate with with other formatting tests

**Review by jreback — CHANGES_REQUESTED** [review]
will also want to add this arg for a datetime index as well. (can be done as a followup or even better in this PR)

**Comment by mroeschke** [comment]
Thanks for the PR, but going to close due the lack of activity. Let us know if you'd like to continue working on this.

**Comment by swt2c** [comment]
I'm planning to take over this PR as the original author seems to have lost time/interest.

File: pandas/tests/scalar/timestamp/test_formats.py (hunk 5 above)
Comment (swt2c) [reply pandas/tests/scalar/timestamp/test_formats.py:4]: Can you suggest a more specific location that would be better?  I looked, but couldn't find an obvious place where other formatting tests are located.

**Pull request closed without merging**